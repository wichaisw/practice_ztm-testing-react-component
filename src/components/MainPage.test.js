import React from 'react';
import { shallow  } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false,
  }

  wrapper = shallow(<MainPage {...mockProps} />);
});

it('should to render MainPage component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should filters robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'Adam',
      email: 'adam@gmail.com'
    }],
    searchField: 'a',
    isPending: false,
  }

  const filteredRobots = [{
    id: 3,
    name: 'Adam',
    email: 'adam@gmail.com'
  }];
  const wrapper2 = shallow(<MainPage {...mockProps2} /> );

  // instance let you have access for all methods in component
  expect(wrapper2.instance().filterRobots(mockProps2.robots)).toEqual(filteredRobots);
});