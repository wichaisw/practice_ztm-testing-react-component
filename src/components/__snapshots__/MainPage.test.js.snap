// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should to render MainPage component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
